cli/cmd/profile.go

```diff
@@ -34,15 +35,38 @@ type profileOptions struct {
 	openAPI   string
 }
 
-func newProfileOptions() *profileOptions {
-	return &profileOptions{
+func newProfileOptions() profileOptions {
+	return profileOptions{
 		name:      "",
 		namespace: "default",
 		template:  false,
 		openAPI:   "",
 	}
 }
 
+func validateOptions(options profileOptions) error {
+	outputs := 0
+	if options.template {
+		outputs++
+	}
+	if options.openAPI != "" {
+		outputs++
+	}
+	if outputs != 1 {
+		return errors.New("You must specify exactly one of --template or --open-api")
+	}
+
+	if msgs := path.IsValidPathSegmentName(options.name); len(msgs) != 0 {
```

Comment:
TIOLI: I was a bit confused by this until I looked up the contract of IsValidPathSegmentName.  Renaming `msgs` to `errs` might make this more clear.